diff --git a/proxy/meson.build b/proxy/meson.build
index 5bdaf7d..516969a 100644
--- a/proxy/meson.build
+++ b/proxy/meson.build
@@ -9,5 +9,3 @@ endif
 if get_option('environment_proxy').enabled()
   subdir('environment')
 endif
-
-subdir('tests')
diff --git a/tls/meson.build b/tls/meson.build
index e76e766..b8103cd 100644
--- a/tls/meson.build
+++ b/tls/meson.build
@@ -13,5 +13,3 @@ if backends.contains('openssl')
 
   subdir('openssl')
 endif
-
-subdir('tests')
