diff --git a/meson.build b/meson.build
index 116f8ba..7415083 100644
--- a/meson.build
+++ b/meson.build
@@ -1,8 +1,9 @@
-project('gst-plugins-good', 'c',
+project('gst-plugins-good', 'c', 'cpp',
   version : '1.22.9',
   meson_version : '>= 0.62',
   default_options : [ 'warning_level=1',
-                      'buildtype=debugoptimized' ])
+                      'buildtype=debugoptimized',
+                      'cpp_std=c++17' ])
 
 gst_version = meson.project_version()
 version_arr = gst_version.split('.')
