diff --git a/meson.build b/meson.build
index ce7638d..581d411 100644
--- a/meson.build
+++ b/meson.build
@@ -2,7 +2,8 @@ project('gst-plugins-bad', 'c', 'cpp',
   version : '1.22.9',
   meson_version : '>= 0.62',
   default_options : [ 'warning_level=1',
-                      'buildtype=debugoptimized' ])
+                      'buildtype=debugoptimized',
+                      'cpp_std=c++17' ])
 
 gst_version = meson.project_version()
 version_arr = gst_version.split('.')
