diff --git a/ext/soup/gstsouploader.c b/ext/soup/gstsouploader.c
index 549b81f..d624aaf 100644
--- a/ext/soup/gstsouploader.c
+++ b/ext/soup/gstsouploader.c
@@ -162,9 +162,22 @@ gst_soup_load_library (void)
       libsoup_sonames[0] = LIBSOUP_2_SONAME;
       GST_DEBUG ("LibSoup 2 found");
     } else {
-      GST_DEBUG ("Trying all libsoups");
-      libsoup_sonames[0] = LIBSOUP_3_SONAME;
-      libsoup_sonames[1] = LIBSOUP_2_SONAME;
+      gchar *libsoup3_library_path = g_getenv ("LIBSOUP3_LIBRARY_PATH");
+      if (libsoup3_library_path) {
+        GST_DEBUG ("Trying %s", libsoup3_library_path);
+        libsoup_sonames[0] = libsoup3_library_path;
+      }
+      else {
+        libsoup_sonames[0] = LIBSOUP_3_SONAME;
+      }
+      gchar *libsoup2_library_path = g_getenv ("LIBSOUP2_LIBRARY_PATH");
+      if (libsoup2_library_path) {
+        GST_DEBUG ("Trying %s", libsoup2_library_path);
+        libsoup_sonames[1] = libsoup2_library_path;
+      }
+      else {
+        libsoup_sonames[1] = LIBSOUP_2_SONAME;
+      }
     }
 
     g_clear_pointer (&handle, dlclose);
@@ -221,6 +234,7 @@ gst_soup_load_library (void)
   len = g_strv_length ((gchar **) libsoup_sonames);
 
   for (guint i = 0; i < len; i++) {
+    GST_DEBUG ("Loading %s", libsoup_sonames[i]);
     module =
         g_module_open (libsoup_sonames[i],
         G_MODULE_BIND_LAZY | G_MODULE_BIND_LOCAL);
